Packs/JWT/Integrations/JWT/JWT.py

```diff
@@ -0,0 +1,188 @@
+import demistomock as demisto
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
+
+import requests
+import traceback
+from typing import Dict
+import jwt
+import uuid
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()  # pylint: disable=no-member
+
+
+''' CONSTANTS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'  # ISO8601 format with UTC, default in XSOAR
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+
+    def request_access_token(self, headers, body):
+        response = self._http_request('post', headers=headers, json_data=body)
+        return response
+
+
+''' HELPER FUNCTIONS '''
+
+
+def encode_authentication_token(secret_key, jti=None, iss=None, aud=None, sub=None, scp=None, iat=None, exp=None,
+                                nbf=None, token_timeout=None, additional_claims=None):
+    token_id = str(uuid.uuid4())
+    jti = jti or token_id
+
+    token_timeout = token_timeout or '300'
+    now = datetime.utcnow()
+    timeout_datetime = now + timedelta(seconds=int(token_timeout))
+    epoch_time = int((now - datetime(1970, 1, 1)).total_seconds())
+    epoch_timeout = int((timeout_datetime - datetime(1970, 1, 1)).total_seconds())
+    exp = exp or epoch_timeout
+    iat = iat or epoch_time
+
+    claims = {
+        'exp': exp,
+        'iat': iat,
+        'jti': jti
+    }
+
+    if iss:
+        claims['iss'] = iss
+    if aud:
+        claims['aud'] = aud
+    if sub:
+        claims['sub'] = sub
+    if scp:
+        claims['iss'] = scp
+    if scp:
+        claims['nbf'] = nbf
+    if additional_claims:
+        claims.update(json.loads(additional_claims))
+
+    payload = jwt.encode(claims, secret_key, algorithm='HS256')
+
+    return jti, payload
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module():
+    return 'ok'
```

Comment:
Why test_module is not checking anything?

Reply:
JWT has different set of claims (jti,iss,aud, ... and unlimited number of additional claims) that largely vary with different authentication services, some services that are to be used with this integration to generate JWT tokens may expect claims in the authentication payload that are not defined in the integration parameters which would result in a test failure because of missing claims, I can create additional parameters for all the RFC claims and have one parameter that accept additional (non RFC claims) as a JSON object, in that case we can have some testing login since we have all the claims needed for the test . will push another commit to reflect that

Reply:
Got it
I think in that case we can just returns 'ok' in the test, because now when user wants to configure the integration he need to populate non mandatory params like Issuer and Subject in order to pass the test.